apps/meteor/app/livechat/server/lib/QueueManager.ts

```diff
@@ -255,7 +257,17 @@ export class QueueManager {
 			void notifyOnSettingChanged(livechatSetting);
 		}
 
-		const newRoom = (await this.queueInquiry(inquiry, room, defaultAgent)) ?? (await LivechatRooms.findOneById(rid));
+		let newRoom;
+		// Note: if the contact is not verified, we cannot add it to the queue, since no agent should be able to pick the conversation
+		//       up. So if the contact is not verified we don't add it to queue and wait an event to be sent to properly add it to queue
+		if (await isUnverifiedContact(room)) {
+			newRoom = await LivechatRooms.findOneById(rid);
+			await unverifyContactChannel(inquiry.v.contactId, room.source.type, inquiry.v._id);
```

Comment:
why unverify if it's already unverified?

Reply:
it might not be, if the setting to Request on first contact only is false
I think I should rename the function to shouldTriggerVerificationApp